source/DistanceMap.h

```diff
@@ -41,7 +41,9 @@ class DistanceMap {
 	// Calculate the path for the given ship to get to the given system. The
 	// ship will use a jump drive or hyperdrive depending on what it has. The
 	// pathfinding will stop once a path to the destination is found.
-	DistanceMap(const Ship &ship, const System *destination);
+	// If a player is given, the path will only include systems that the
+	// player has visisted.
+	DistanceMap(const Ship &ship, const System *destination, const PlayerInfo *player = nullptr);
```

Comment:
Probably we should refactor `DistanceMap` to take an optional set of known systems, rather than a player object.

(We can do that in a sooner-rather-then-later follow-up though.)

Reply:
We need the player object as well to check if a wormhole was visited by the player.

Reply:
That's required by the current implementation, yes. If we're passing in the knowledge of usable systems, passing in the knowledge of usable planets/wormholes is probably expected.